quantfinlib/train_test_split/kfoldpe.py

```diff
@@ -0,0 +1,197 @@
+"""Implementation of time-aware KFold Cross Validation with Purge and Embargo."""
+
+from numbers import Integral
+from typing import Generator, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+
+from quantfinlib.train_test_split._base import BaseCV, _purge, _embargo
+
+
+def _validate_split_settings(n_samples: int, n_folds: int, n_embargo: int, n_purge: int) -> None:
+    """Validate the split settings."""
+    if n_samples < n_folds:
+        raise ValueError("Number of samples must be greater than the number of folds.")
+    if n_purge + n_embargo >= n_samples:
+        raise ValueError(f"The sum {n_purge + n_embargo = } must be less than {n_samples = }.")
+    max_train_size = n_samples - n_samples // n_folds
+    if n_purge > max_train_size:
+        raise ValueError(f"{n_purge = } must be less than {max_train_size = }.")
+    if n_embargo > max_train_size:
+        raise ValueError(f"{n_embargo = } must be less than {max_train_size = }.")
+    if n_purge + n_embargo > max_train_size:
+        raise ValueError(f"The sum {n_purge + n_embargo = } must be less than {max_train_size = }.")
+
+
+class TimeAwareKFold(BaseCV):
+    """Purge-embargo K-Fold cross-validator.
+
+    Provides train/test indices to split data in train/test sets. The training set is purged and embargoed.
+
+    Attributes
+    ----------
+    n_folds : int

```

Comment:
n_folds parameter, but then you use `splits` later on in varialbe names and docstrings. also WalkForward uses `splits` notation.

Reply:
I made it clear in the docstring. n_splits is n_folds if look_forward is False, otherwise it is n_folds - 1 because we remove discard the train-test split for which the test set is behind the train set time-wise.